src/main/java/org/pentaho/di/trans/dataservice/resolvers/DataServiceResolverDelegate.java

```diff
@@ -0,0 +1,115 @@
+ *
+ ******************************************************************************/
+
+package org.pentaho.di.trans.dataservice.resolvers;
+
+import com.google.common.base.Function;
+import org.pentaho.di.core.exception.KettleException;
+import org.pentaho.di.core.logging.LogChannelInterface;
+import org.pentaho.di.core.sql.SQL;
+import org.pentaho.di.trans.dataservice.DataServiceExecutor;
+import org.pentaho.di.trans.dataservice.DataServiceMeta;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataServiceResolverDelegate implements DataServiceResolver {
+
+  private List<DataServiceResolver> resolvers;
+
+  public DataServiceResolverDelegate( List<DataServiceResolver> resolvers ) {
+    this.resolvers = resolvers;
+  }
+
+  @Override public List<DataServiceMeta> getDataServices( Function<Exception, Void> logger ) {
+    List<DataServiceMeta> dataServiceMetas = new ArrayList<>();
+    for ( DataServiceResolver resolver : resolvers ) {
+      dataServiceMetas.addAll( resolver.getDataServices( logger ) );
+    }
+    return dataServiceMetas;
+  }
+
+  @Override public List<DataServiceMeta> getDataServices( String dataServiceName, Function<Exception, Void> logger ) {
+    List<DataServiceMeta> dataServiceMetas = new ArrayList<>();
+    for ( DataServiceResolver resolver : resolvers ) {
+      dataServiceMetas.addAll( resolver.getDataServices( dataServiceName, logger ) );
+    }
+    return dataServiceMetas;
+  }
+
+  @Override public DataServiceMeta getDataService( String dataServiceName ) {
+    for ( DataServiceResolver resolver : resolvers ) {
+      DataServiceMeta dataServiceMeta = resolver.getDataService( dataServiceName );
+      if ( dataServiceMeta != null ) {
+        return dataServiceMeta;
+      }
+    }
+    return null;
+  }
+
+  @Override public List<String> getDataServiceNames( String dataServiceName ) {
+    List<String> dataServiceNames = new ArrayList<>();
+    for ( DataServiceResolver resolver : resolvers ) {
+      dataServiceNames.addAll( resolver.getDataServiceNames( dataServiceName ) );
+    }
+    return dataServiceNames;
+  }
+
+  @Override public DataServiceExecutor.Builder createBuilder( SQL sql ) throws KettleException {
+    for ( DataServiceResolver resolver : resolvers ) {
+      DataServiceExecutor.Builder builder = resolver.createBuilder( sql );
+      if ( builder != null ) {
+        return builder;
+      }
+    }
+    throw new KettleException( "Error when creating builder for sql query" );
+  }
+
+  @Override public List<String> getDataServiceNames() {
+    List<String> dataServiceNames = new ArrayList<>();
+    for ( DataServiceResolver resolver : resolvers ) {
+      dataServiceNames.addAll( resolver.getDataServiceNames() );
+    }
+    return dataServiceNames;
+  }
+
+  @Override public LogChannelInterface getLogChannel() {
+    for ( DataServiceResolver resolver : resolvers ) {
+      LogChannelInterface logChannel = resolver.getLogChannel();
+      if ( logChannel != null ) {
```

Comment:
We only return the first log channel found.  Is that a problem?  Below it looks like the transientResolver has a null LogChannelInterface... how do we handle logging for transients?